tests/acceptance/nested-preprocessor-test-slow.js

```diff
@@ -0,0 +1,332 @@
+    } else {
+      localMarker += '-removed-preprocessTree';
+      localMarker += ' /' + relativePath;
+      addon._findHost().project.ui.writeLine(localMarker);
+      return content.replace(localPreprocessTreeMarker, localMarker);
+    }
+  });
+
+  return tree;
+};
+
+describe('Acceptance: nested preprocessor tests.', function() {
+  this.timeout(1000 * 60 * 10);
+  var root;
+
+  before(function() {
+    root = new AppFixture('root');
+
+    var types = ['js', 'css', 'template'];
+    var extensions = ['js', 'css', 'hbs'];
+
+    var name, type, ext;
+    var inRepoAddons = {};
+    var nestedInRepoAddons = {};
+    for (var i = 0; i < types.length; i++) {
+      type = types[i];
+      ext = extensions[i];
+      name = type + '-addon';
+
+      inRepoAddons[type] = new InRepoAddonFixture(name);
+      inRepoAddons[type].addMethod('preprocessTree', preprocessTreeFixture.toString());
+      inRepoAddons[type].addMethod('postprocessTree', postprocessTreeFixture.toString());
+      if (type !== 'css') {
+        inRepoAddons[type].addMethod('setupPreprocessorRegistry', generatePreprocessor({
+          registryType: type,
+          ext: ext
+        }));
+      }
+
+      name = name + '-nested';
+      nestedInRepoAddons[type] = new InRepoAddonFixture(name);
+      nestedInRepoAddons[type].addMethod('preprocessTree', preprocessTreeFixture.toString());
+      nestedInRepoAddons[type].addMethod('postprocessTree', postprocessTreeFixture.toString());
+      if (type !== 'css') {
+        nestedInRepoAddons[type].addMethod('setupPreprocessorRegistry', generatePreprocessor({
+          registryType: type,
+          ext: ext
+        }));
+      }
+    }
+
+    var child = new InRepoAddonFixture('child-addon');
+
+    inRepoAddons['css'].generateCSS('app/styles/app.css');
+    inRepoAddons['css'].generateCSS('app/styles/addon.css');
+    inRepoAddons['css'].generateCSS('app/styles/_import.css');
+    inRepoAddons['css'].generateCSS('app/styles/' + inRepoAddons['css'].name + '.css');
+    inRepoAddons['css'].generateCSS('app/styles/alpha.css');
+    inRepoAddons['css'].generateCSS('app/styles/zeta.css');
+    inRepoAddons['css'].generateCSS('addon/styles/addon.css');
+    inRepoAddons['css'].generateCSS('addon/styles/app.css');
+    inRepoAddons['css'].generateCSS('addon/styles/_import.css');
+    inRepoAddons['css'].generateCSS('addon/styles/' + inRepoAddons['css'].name + '.css');
+    inRepoAddons['css'].generateCSS('addon/styles/alpha.css');
+    inRepoAddons['css'].generateCSS('addon/styles/zeta.css');
+
+    inRepoAddons['js']._npmAddonInstall({ name: 'ember-cli-babel' });
+    inRepoAddons['js'].generateJS('app/routes/hat.js');
+    inRepoAddons['js'].generateJS('app/routes/cat.js');
+    inRepoAddons['js'].generateJS('addon/components/thing-one.js');
+    inRepoAddons['js'].generateJS('addon/components/thing-two.js');
+
+    inRepoAddons['template']._npmAddonInstall({ name: 'ember-cli-htmlbars' });
```

Comment:
I'm working around the limitations of the API I have so far. `install` is used to approximate `ember install your-ember-addon` and delegates to underscored functions. You would pass it a fixture which is either an in-repo-addon, or a "full" addon. A "full" addon gets "installed" into the `node_modules` folder, the `_npmAddonInstall` method name is to distinguish from `_inRepoAddonInstall`

What I needed here was to modify the fixture to add a new dependency. I elected to leverage something that I already knew did it rather than add a new helper function, or do it manually in multiple locations because I'm not sure what that API should look like yet.